### PR TITLE
Use TScreen.MenuFont instead of Canvas.Font in TJvMainMenu and TJvPopupMenu

### PR DESCRIPTION
**Pull request by MHumm**
As proposed in mantis issue 6548
[http://issuetracker.delphi-jedi.org/view.php?id=6548](url)

### CODE REVIEW

File: jvcl/run/JvMenus.pas [hunk 1]
```diff
@@ -1255,7 +1255,7 @@ procedure TJvMainMenu.WMDrawItem(var Msg: TWMDrawItem);
         SaveIndex := SaveDC(hDC);
         try
           Canvas.Handle := hDC;
-          SetDefaultMenuFont(Canvas.Font);
+          SetDefaultMenuFont(TScreen.MenuFont);
```
Comment (obones) [inline jvcl/run/JvMenus.pas]: Does this compile with all Delphi versions? I'm wondering if this was available in D6 for instance

**Comment by MHumm** [comment]
I don't know if it compiles with D6. I had at least checked as far back as XE2 or so via TScreen documentation in EMBT DocWiki. Question: who's got D6 installed and can check if that property exists? Or who can tell us when it was introduced if not present in D6 already?

**Comment by MHumm** [comment]
Question of existence in D6 raised here (in German):
[https://www.delphipraxis.net/200519-gibt-es-tscreen-menufont-schon-d6.html#post1431149](url)

**Comment by obones** [comment]
I have a VM at work with all versions of Delphi from D6 to Rio installed, I'll check on Monday

**Comment by MHumm** [comment]
No need to. Delphipraxis folks already confirmed it is available in D6 already. Will fix it later and resubmit.

File: jvcl/run/JvMenus.pas (hunk 1 above)
Comment (obones) [reply jvcl/run/JvMenus.pas]: Well, ok, `MenuFont` exists on `TScreen` but not as a class property, the code should really be `Screen.MenuFont` (notice the missing T at the beginning)

**Comment by MHumm** [comment]
Closed this pull request as it is superceeded by #110 with the fixed version.

**Comment by ronaldhoek** [comment]
@MHumm:  Markus, you do not need to create a new pull request to submit changes to a pull with commit. Just push changes to the branch you used to create the pull request and those will be included into the original pull. 

@obones: or is this contrary to the way pull requests are desired?

_Note_
Basically, a pull request is just a request to merge a branch. Updating the branch (with extra commits) will just result in a merge all of these extra changes including the previously committed changes (at the time you create the pull).
Of course, when the extra changes are so vast, you might consider creating a new branch and then commit the change to that new branch and create a new pull.

**Comment by MHumm** [comment]
@ronaldhoek: Thanks for the tip, I'm still a bit new to Git. One doubt I have with this:
I create a branch, modify some source, commit and push. Then I create the pull request ok. 
Now something comes up which makes me want to change another line of code. 

If I activate that branch again do the modification and commit and push it, what will happen? As far as I understand you it would land in the same pull request, which would be fine for me, but it would be a 2nd commit in that request. Correct? If yes, then that's not what @obones  likes, as he likes us to create pull requests containing only a single commit as it makes his life easier.

=> I'd like to have further guidance on that subject in order to be able to effectively contribute. Until then I'll create new pull requests for changed submissions.

**Comment by ronaldhoek** [comment]
> but it would be a 2nd commit in that request. Correct? 
Yes

>  then that's not what @obones likes
I don't know how he actually looks at the changes (per commit of per file).
What I do know is that @obones wants to see a pull request (= branch) per issue.
I'm not sure whether multiple commits are a problem, as the review of change are probably on a 'per file' basis and the changes of multiple commits are shown merged on Github.

> it makes his life easier
A good point of view - so @obones should give his two cents on this.

**Comment by jfudickar** [comment]
I think the comment from Olivier is not to put two differnt topics into one merge request.
Having a fix done in multiple commits on the same topic should be fine.
For example, fixing a naming or formatting issue.

**Comment by MHumm** [comment]
Ok, if that's the case I can fix my pull requests with new commits in the future. I just don't want obones to reject them because they contain several commits to the same topic. 
Separating topics by pull requests is desirable of course!

**Comment by obones** [comment]
Yes, that is what I meant, one issue = one pull request
I would prefer if there is a single commit as it makes things easier to review. BUT, and that's the exception, if the review requires changes to be done, I'm completely fine if they are placed in a commit in that same branch the PR is linked to.
It even makes it clear that the new commit is in reaction to the review comment.

**Pull request closed without merging**